datumaro/plugins/transforms.py

```diff
@@ -734,3 +736,82 @@ def transform_item(self, item):
                 label=bbox.label, attributes=bbox.attributes))
 
         return item.wrap(annotations=annotations)
+
+class ResizeTransform(ItemTransform):
+    """
+    Resizes images and annotations in the dataset to the specified size.
+    Supports upscaling, downscaling and mixed variants.|n
+    |n
+    Examples:|n
+    - Resize all images to 256x256 size|n
+    |s|s%(prog)s -dw 256 -dh 256
+    """
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument('-dw', '--width', type=int,
+            help="Destination image width")
+        parser.add_argument('-dh', '--height', type=int,
+            help="Destination image height")
+        return parser
+
+    def __init__(self, extractor: IExtractor, width, height) -> None:
+        super().__init__(extractor)
+
+        assert width > 0 and height > 0
+        self._width = width
+        self._height = height
+
+    def transform_item(self, item):
+        if not item.has_image:
+            raise Exception("Image info is required for this transform")
+
+        h, w = item.image.size
+        xscale = self._width / float(w)
+        yscale = self._height / float(h)
+
+        if item.image.has_data:
+            method = cv2.INTER_LINEAR if (xscale * yscale) < 1 \
+                else cv2.INTER_LANCZOS4
+            image = item.image.data / 255.0
+            resized_image = cv2.resize(image, (0, 0),
```

Comment:
I think it would be slightly better to pass in the destination dimensions and omit `fx` and `fy`:

1. It guarantees that the result will be of the exact desired dimensions (not affected by rounding weirdness).
2. You have to pass in the dimensions anyway, so might as well pass the correct ones.

Reply:
Added